python/lognplot/qt/render/chart.py

```diff
@@ -77,17 +82,17 @@ def _draw_curve(self, curve):
 
         if data:
             if isinstance(data[0], Aggregation):
-                self._draw_aggregations_as_shape(data, curve_color)
+                curve.average = self._draw_aggregations_as_shape(curve.axis, data, curve_color)
```

Comment:
You could query the average from by querying the summary of the data in view to the time series database (tsdb). This will give you a quicker result than re-calculating the average in the draw function.

Also: having the draw function doing two things (draw the plot and calculate the average) is probably not so clean, it would be better to seperate the two functions.

Reply:
Sure, still have to get familiar with the codebase.